### PR TITLE
Break out the code into files that each represent a single idea

### PR DESCRIPTION
**Pull request by chee**
Mostly moves code around, but does make a few more significant changes.

## More significant changes

You might be wondering about the "more significant changes" mentioned in the previous and only sentence in the PR. Here's an abridged unordered, yet numbered, list:

1. [Declare that we are using Rust 2018 edition](https://github.com/Financial-Times/oax/blob/87992da96a9983744f5edebf543c1cb6a34b7bd0/Cargo.toml#L5), and update the code to reflect that (that is: remove `extern crate` from `main.rs`)
1. Remove the `--es-syntax` and `--es-syntax-everywhere` options because we _always_ want those on. same for the `browser` option.
1. [Replace the `for_bower` boolean with a `PackageManager` enum, move the functions for getting the component dir and manifest file name to that enum's implementation](https://github.com/Financial-Times/oax/blob/87992da96a9983744f5edebf543c1cb6a34b7bd0/src/input_options.rs#L9-L23)
1. add some helpful comments, and some TODO comments

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,14 +1,15 @@
 [package]
 name = "oax"
-version = "0.4.0"
-authors = ["Nathan"]
+version = "0.5.0"
+authors = ["Nathan", "chee", "Jake"]
```
Comment (JakeChampion) [inline Cargo.toml:4]: <3

File: Cargo.toml [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+edition = "2018"
 
-description = "The fastest JavaScript bundler in the galaxy."
-repository = "https://github.com/nathan/pax"
+description = "A fast JavaScript bundler for Origami components"
+repository = "https://github.com/financial-times/oax"
```
Comment (JakeChampion) [inline Cargo.toml:8]: <3

File: src/input_options.rs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+use fnv::FnvHashSet;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum PackageManager {
+    Bower,
+    Npm,
+}
+
+impl PackageManager {
```
Comment (JakeChampion) [inline src/input_options.rs:9]: This is really nice

File: src/javascript/tail.js [hunk 4]
```diff
@@ -0,0 +1 @@
+})(typeof global !== "undefined" ? global : typeof window !== "undefined" ? window : this)
```
Comment (JakeChampion) [inline src/javascript/tail.js:1]: TODO: add `globalThis` for modern browsers and `self` for web workers in older browsers

Comment (chee) [reply src/javascript/tail.js:1]: ++

File: src/vlq.rs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+use std::str;
```
Comment (JakeChampion) [inline src/vlq.rs:1]: TODO: maybe use https://crates.io/crates/vlq instead?

**Review by JakeChampion — APPROVED** [review]

**Pull request merged**